shotover-proxy/tests/runner/hotreload_int_tests.rs

```diff
@@ -1,10 +1,81 @@
 use crate::shotover_process;
+use redis::{Client, Commands};
+use std::time::Duration;
+use test_helpers::docker_compose::docker_compose;
+use tokio::time::sleep;
+
 #[tokio::test]
-async fn test_hotreload_cli_flag_accepted() {
-    let shotover_process = shotover_process("tests/test-configs/null-valkey/topology.yaml")
+async fn test_hotreload_basic_redis_connection() {
+    let _compose = docker_compose("tests/transforms/docker-compose-moto.yaml");
+    sleep(Duration::from_millis(2000)).await;
+    let shotover_process = shotover_process("tests/test-configs/redis-passthrough/topology.yaml")
         .with_hotreload(true)
         .with_config("tests/test-configs/shotover-config/config_metrics_disabled.yaml")
         .start()
         .await;
+    sleep(Duration::from_millis(1000)).await;
+    let client = Client::open("redis://127.0.0.1:6380").expect("Failed to create Redis client");
+    let mut con = client
+        .get_connection()
+        .expect("Failed to connect to Redis through shotover");
+
+    let _: () = con
+        .set("test_key", "test_value")
+        .expect("Failed to SET key");
+    let result: String = con.get("test_key").expect("Failed to GET key");
+    assert_eq!(result, "test_value");
+    let pong: String = redis::cmd("PING").query(&mut con).expect("Failed to PING");
+
+    assert_eq!(pong, "PONG");
```

Comment:
nit: this will be easier to read like:

```
    let pong: String = redis::cmd("PING").query(&mut con).expect("Failed to PING");
    assert_eq!(pong, "PONG");

    shotover_process.shutdown_and_then_consume_events(&[]).await;
```

Makes it clear that line 29 is related to line 27, not line 30.